src/cljs/snapshot/lumo/repl.cljs

```diff
@@ -646,6 +646,57 @@
   "Wrap wfn around all (fn) values in fns hashmap."
   (into {} (for [[k v] fns] [k (wfn v)])))
 
+;; --------------------
+;; REPL upgrade
+
+(deftype ^:private SuspensionRequest [f])
+
+(defn suspension-request? [x] (instance? SuspensionRequest x))
+
+(defn suspension-request [f] (SuspensionRequest. f))
+
+(defprotocol AsyncReader
+  "Asynchronous stream of strings."
+  (read-chars [r f] "Calls f with a string or nil (EOF)")
+  (pushback [r s] "Unread s"))
+
+(defn yield-control [suspension-request async-reader resume-cb]
+  ((.-f suspension-request) async-reader resume-cb))
+
+(defn- create-async-pipe []
+  (let [front #js []
+        back #js []
+        cb (volatile! nil)
+        spill! #(when-some [s (.pop back)] (.push front s) (recur))]
+    #js [(fn 
```

Comment:
I mean this line: 

```
(fn 
        ([]
         (spill!)
```
The two-arity we, minor anyways

Reply:
I added comments to each arity in 8f8eec4e469de4a5a6727cfb1ed3e260bf51e66b